docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+- Provide good defaults where possible
+
+In Java, the builder pattern is common for configuring instances. Let's look at what that may look
+like. The simplest configuration will be when a user wants to get a default experience, exporting
+with a specific exporter to an endpoint.
+
+```java
+OpenTelemetrySdk openTelemetry = OpenTelemetrySdk.builder()
+  .addSpanExporter(OtlpGrpcSpanExporter.builder()
+    .setEndpoint("spans-service:4317")
+    .build())
+  .addMetricExposition(PrometheusExposition.getDefault())
+  .build();
+```
+
+This
+
+- Exports spans using OTLP to spans-service
+  - Uses the BatchSpanProcessor (or a batching exporter wraper) which we consider the best default 
+  for exporting. It has defaults for queue size, timeout, etc
+- Exposes metrics using Prometheus format
+- Uses ParentBased(AlwaysOn) sampler
+- Uses standard random IDs
+- Uses a default Resource which simply consists of the SDK (or any other resources we decide to)
+include in the core SDK, not extensions
+- Uses the default Clock, which uses Java 8 / 9+ optimized APIs for getting time
+  - The only real reason a user would set this is for unit tests, not for production
+- Enforces default numeric limits related to number of attribute, etc
+
+Because the exporting is often the only aspect an end user needs to configure, this is the simplest
+possible API for configuring the SDK.
+
+Let's look at a more complicated example
+
+```java
+OpenTelemetrySdk openTelemetry = OpenTelemetrySdk.builder()
+  .setResource(Resource.getDefault().merge(CoolResource.getDefault()))
+  .setClock(AtomicClock.create())
+  .setPropagators(B3Propagator.getInstance())
+  .addSpanExporter(OtlpGrpcSpanExporter.builder()
+    .setEndpoint("spans-service:4317")
+    .setTimeout(Duration.ofSeconds(10))
+    .setBatchQueueSize(10000)
+    .build())
+  .addSpanExporter(ZipkinExporter.builder()
+    .setEndpoint("spans-service:4317")
+    .disableBatching()
+    .build())
+  .addSpanExporter(FullyAsyncZipkinExporter.create())
+  .addSpanProcessor(CustomAttributeAddingProcessor.create())
+  .addSpanProcessor(CustomEventAddingProcessor.create())
+  .setTraceSampler(Sampler.rateLimiting())
+  .setTraceLimits(TraceLimits.builder().setMaxAttributes(10).build())
+  .setIdGenerator(TimestampedIdGenerator.create())
+  .setMeterStuff(InProgressMetricsStuff.create())
+  .build()
+```
+
+This configures resource, clock, exporters, span processors, sampler, trace limits, and metrics.
+Everything is exposed on the `OpenTelemetrySdkBuilder` to provide discoverability of all the parts
+of the SDK and to let the SDK manage shared components - it is not possible to configure a different
+`Resource` for tracing and metrics, something that would be a gotcha. Exporters are handled natively
+given they are so important - even though the exporters are specified before span processors, they
+will always be executed last to make sure all span processors appy to them. The exporter builders
```

Comment:
How do you plan to chain `SpanProcessor.end()` calls?

Reply:
This proposal doesn't address the issue of not being able to have branching `SpanProcessor`. It just calls the processors in order, but trying to help with the exporter gotcha by making sure it comes at the end.

Reply:
This seems to be as violating spec, et least formally